support/routes/v2/isobot/ops/botService.ts

```diff
@@ -0,0 +1,97 @@
+import { createHmac, timingSafeEqual } from "node:crypto"
+
+import qs from "qs"
+
+import { config } from "@config/config"
+
+import logger from "@root/logger/logger"
+import WhitelistService from "@root/services/identity/WhitelistService"
+
+export interface SlackPayload {
+  token: string
+  team_id: string
+  team_domain: string
+  channel_id: string
+  channel_name: string
+  user_id: string
+  user_name: string
+  command: string
+  text: string
+}
+
+class BotService {
+  whitelistService: WhitelistService
+
+  constructor(whitelistService: WhitelistService) {
+    this.whitelistService = whitelistService
+  }
+
+  public verifySignature(
+    signature: string,
+    timestamp: string,
+    payload: SlackPayload
+  ) {
+    // Verify timestamp is within 5 minutes
+    const currentTime = new Date().getTime() / 1000
+    const slackTimestamp = parseInt(timestamp)
+
+    if (Math.abs(currentTime - slackTimestamp) > 300) {
+      return false
+    }
+
+    // Create HMAC with the signing secret and the timestamp
+    const signingSecret = config.get("slackbot.secret")
+    const versionNumber = "v0"
+    const reqBody = qs.stringify(payload, { format: "RFC1738" })
+    const signatureBasestr = `${versionNumber}:${timestamp}:${reqBody}`
+    const computedHash = createHmac("sha256", signingSecret)
+      .update(signatureBasestr)
+      .digest("hex")
+    const computedSig = `${versionNumber}=${computedHash}`
+
+    logger.info({
+      message: `Slack verification metadata`,
+      meta: {
+        currentTime,
+        slackTimestamp,
+        reqBody,
+      },
+    })
+
+    if (timingSafeEqual(Buffer.from(signature), Buffer.from(computedSig))) {
+      logger.info({ message: "Signature verified" })
+      return true
+    }
+    logger.error({ message: "Signature verification failed" })
+    return false
```

Comment:
something we could do here is return a `VerifiedSignature`, where `VerifiedSignature = Tagged<"verified">` (see [here](https://github.com/sindresorhus/type-fest/blob/main/source/opaque.d.ts)). 

this is an example of parse, don't validate (blog post [here](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)), which encodes this idea of "validity" into our typesystem and prevents duplicate work

Reply:
not doing - we'll deprecate this method cos bolt does it for us